lib/storage/data/external/AzureClient.js

```diff
@@ -1,62 +1,77 @@
-const url = require('url');
-
-const azure = require('azure-storage');
+const { BlobServiceClient, StorageSharedKeyCredential } = require('@azure/storage-blob');
 const errors = require('../../../errors').default;
 const azureMpuUtils = require('../../../s3middleware/azureHelpers/mpuUtils');
 const { validateAndFilterMpuParts } =
     require('../../../s3middleware/processMpuParts');
 
 const { createLogger, logHelper, translateAzureMetaHeaders } =
     require('./utils');
+const objectUtils = require('../../../s3middleware/objectUtils');
 
 const constants = require('../../../constants');
 const packageVersion = require('../../../../package.json').version;
 
-azure.Constants.USER_AGENT_PRODUCT_NAME = constants.productName;
-azure.Constants.USER_AGENT_PRODUCT_VERSION = packageVersion;
-
 class AzureClient {
     constructor(config) {
         this._azureStorageEndpoint = config.azureStorageEndpoint;
         this._azureStorageCredentials = config.azureStorageCredentials;
         this._azureContainerName = config.azureContainerName;
-        this._client = azure.createBlobService(
+        const cred = new StorageSharedKeyCredential(
             this._azureStorageCredentials.storageAccountName,
             this._azureStorageCredentials.storageAccessKey,
-            this._azureStorageEndpoint);
-        this._client.enableGlobalHttpAgent = true;
+        );
+        const proxyOptions = (() => {
+            if (!config.proxy || !config.proxy.url) {
+                return undefined;
+            }
+            // NOTE: config.proxy.certs is not supported
+            const parsedUrl = new URL(config.proxy.url);
+            return {
+                host: parsedUrl.host,
+                port: parsedUrl.port || 80,
+                username: parsedUrl.username || undefined,
+                password: parsedUrl.password || undefined,
+            };
+        })();
```

Comment:
Why is a lambda need for this?
```suggestion
        let proxyOptions;
        if (config.proxy && config.proxy.url) {
            // NOTE: config.proxy.certs is not supported
            const parsedUrl = new URL(config.proxy.url);
            proxyOptions = {
                host: parsedUrl.host,
                port: parsedUrl.port || 80,
                username: parsedUrl.username || undefined,
                password: parsedUrl.password || undefined,
            };
        }
```

Reply:
It is not _needed_, there are obviously other ways to write the code. But it allows keeping the variable `const` with a single initialization, to better convey intention and avoid mis-assignments, all the while keeping the all logic in the same place.